File: algorithms/python/binaryStringWithSubstringsRepresenting1ToN/binaryStringWithSubstringsRepresenting1ToN.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/binary-string-with-substrings-representing-1-to-n/
# Author: Miao Zhang
# Date:   2021-04-02

class Solution:
    def queryString(self, S: str, N: int) -> bool:
        for x in range(1, N + 1):
            if S.find("{0:b}".format(x)) == -1:
                return False
        return True

File: algorithms/cpp/binaryStringWithSubstringsRepresenting1ToN/binaryStringWithSubstringsRepresenting1ToN.cpp
// Source: https://leetcode.com/problems/binary-string-with-substrings-representing-1-to-n/
// Author: Miao Zhang
// Date:   2021-04-02

class Solution {
public:
    bool queryString(string S, int N) {
        for (int i = N; i > 0; i--) {
            string b = bitset<32>(i).to_string();
            if (S.find(b.substr(b.find("1"))) == string::npos) {
                return false;
            }
        }
        return true;
    }
};
